spark/src/main/scala/geotrellis/spark/io/hadoop/geotiff/HadoopIMGeoTiffAttributeStore.scala

```diff
@@ -1,13 +1,28 @@
 package geotrellis.spark.io.hadoop.geotiff
 
+import geotrellis.spark.io.hadoop.HdfsUtils
+
 import org.apache.hadoop.conf.Configuration
+import org.apache.commons.io.IOUtils
+import org.apache.hadoop.fs.Path
+import spray.json._
+import spray.json.DefaultJsonProtocol._
+
 import java.net.URI
 
 object HadoopIMGeoTiffAttributeStore {
   def apply(
     name: String,
     uri: URI,
     conf: Configuration = new Configuration()
-  ): InMemoryGeoTiffAttributeStore =
-    InMemoryGeoTiffAttributeStore(() => GeoTiffMetadataTree.fromGeoTiffMetadataList(HadoopGeoTiffInput.list(name, uri, conf)))
+  ): InMemoryGeoTiffAttributeStore = {
+    def getDataFunction = () => HadoopGeoTiffInput.list(name, uri, conf)
```

Comment:
Kind of strange that we're mixing in anonymous trait here to define `getDataFunction`. We already have `HadoopIMGeoTiffAttributeStore` name, why not make it a boring class that just extends `HadoopIMGeoTiffAttributeStore`?